### PR TITLE
CO-123: harden hot-suite dist freshness heuristics

### PR DESCRIPTION
**Pull request by Kbediako**
## Summary
- add a shared test-local `shouldUseFreshDist` helper that walks each hot suite's transitive relative runtime-import closure instead of relying on narrow hand-maintained roots
- harden the CLI and review hot suites with focused stale-dependency, unrelated-sibling, and unresolved-to-resolvable cache-refresh coverage
- add the CO-123 docs-first packet and mirror updates that record the bounded dependency-root contract and clean review evidence

## Why
CO-114 already preserved the fast retained-subprocess smoke path in the protected CI lane, but local or focused reruns could still treat stale `dist` as fresh when a transitive CLI/runtime dependency changed outside the previously tracked roots.

## Testing
- `npx vitest run tests/cli-command-surface.spec.ts tests/run-review.spec.ts`
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint`
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `node scripts/diff-budget.mjs`
- `FORCE_CODEX_REVIEW=1 CODEX_REVIEW_NON_INTERACTIVE=1 npm run review -- --manifest ../../.runs/linear-63e7f712-d927-43fb-8afa-b357bab40aa2/cli/2026-04-10T06-49-36-719Z-253d7dad/manifest.json`
- `npm run pack:smoke`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved build artifact freshness validation to properly detect when transitive CLI/runtime dependencies have changed, preventing stale pre-built distributions from being incorrectly reused.

* **Documentation**
  * Added comprehensive documentation for build freshness improvements including technical specifications, action plans, and task tracking.

* **Tests**
  * Enhanced test coverage for build freshness validation scenarios, including transitive dependency detection and edge cases.

### CODE REVIEW

File: tests/helpers/distFreshness.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+      if (!resolvedDependency) {
+        hasUnresolvedTrackedDependency = true;
+        for (const candidate of unresolvedCandidates) {
+          trackedPaths.add(candidate);
+        }
+        continue;
+      }
+      if (!files.has(resolvedDependency)) {
+        pending.push(resolvedDependency);
+      }
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tests/helpers/distFreshness.ts:114]: Addressed in commit `6d776d6ac`: the cache now tracks resolution-priority candidates for resolved imports, so a later-added higher-priority source file invalidates the cached closure and re-evaluates freshness. Added a focused regression in `tests/cli-command-surface.spec.ts` covering the late `*.js` candidate case, and reran focused + full validation plus standalone review cleanly.

**Comment by Kbediako** [comment]
Closing for CO-123 rework reset. I am restarting this issue from a fresh origin/main branch/workpad per the provider-worker rework contract.
